Aplib.Core/LiftingExtensionMethods.cs

```diff
@@ -0,0 +1,93 @@
+    /// </summary>
+    public static class LiftingExtensionMethods
+    {
+        /// <summary>
+        /// Wraps a normal action into a tactic.
+        /// </summary>
+        /// <returns>
+        /// A primitive tactic, whose guard always returns true.
+        /// If the action implements <see cref="IDocumented"/>, it's name will be prefixed with an indicator that it
+        /// has been lifted.
+        /// </returns>
+        /// <param name="action">
+        /// The action which on its own can function as a tactic. Meaning, the tactic consists of just a single action.
+        /// </param>
+        /// <param name="metadata">Optional metadata to be assigned to the tactic.</param>
+        public static PrimitiveTactic<TBeliefSet> Lift<TBeliefSet>(
+            this IAction<TBeliefSet> action,
+            IMetadata? metadata = null)
+            where TBeliefSet : IBeliefSet => metadata is null
+                ? new PrimitiveTactic<TBeliefSet>(action: action)
+                : new PrimitiveTactic<TBeliefSet>(metadata, action: action);
+
+        /// <summary>
+        /// Wraps a queryable action into a tactic.
+        /// </summary>
+        /// <returns>
+        /// A primitive tactic, whose guard always returns true.
+        /// If the action implements <see cref="IDocumented"/>, it's name will be prefixed with an indicator that it
+        /// has been lifted.
+        /// </returns>
+        /// <param name="action">
+        /// The action which on its own can function as a tactic. Meaning, the tactic consists of just a single action.
+        /// </param>
+        /// <param name="metadata">Optional metadata to be assigned to the tactic.</param>
+        public static PrimitiveTactic<TBeliefSet> Lift<TBeliefSet>(
+            this IQueryable<TBeliefSet> action,
+            IMetadata? metadata = null)
+            where TBeliefSet : IBeliefSet => metadata is null
+                ? new PrimitiveTactic<TBeliefSet>(queryAction: action)
+                : new PrimitiveTactic<TBeliefSet>(metadata, queryAction: action);
+
+        /// <summary>
+        /// Wraps a goal into a goal structure.
+        /// </summary>
+        /// <returns>
+        /// A primitive goal structure.
+        /// If the goal implements <see cref="IDocumented"/>, it's name will be prefixed with an indicator that it
+        /// has been lifted.
+        /// </returns>
+        /// <param name="goal">
+        /// The goal which on its own can function as a goal structure. Meaning, the goal structure consists of just a
+        /// single goal.
+        /// </param>
+        /// <param name="metadata">Optional metadata to be assigned to the goal structure.</param>
+        public static PrimitiveGoalStructure<TBeliefSet> Lift<TBeliefSet>(
+            this IGoal<TBeliefSet> goal,
+            IMetadata? metadata = null)
+            where TBeliefSet : IBeliefSet => metadata is null
```

Comment:
If I am correct, the metadata will never be null, as a default metadata will be set. Therefore, this ternary boolean expression is not needed, and metadata can just be passed.